scripts/object_storage_cleaner/storage_cleaner.py

```diff
@@ -0,0 +1,628 @@
+
+    return f"{size:.2f} PB"
+
+
+# Main class
+class ObjectStorageCleaner:
+    BATCH_SIZE = 1000
+
+    def __init__(
+        self,
+        bucket: str,
+        region_name: str | None = None,
+        profile: str | None = None,
+        endpoint_url: str | None = None,
+        prefix: str | None = None,
+        deleted_after: datetime | None = None,
+    ):
+        self.bucket = bucket
+        self.prefix = prefix
+        self.deleted_after = deleted_after
+        self.region_name = region_name
+        self.profile = profile
+        self.endpoint_url = endpoint_url
+
+        # Metrics
+        self._scanned_count = 0
+        self._stats_keys = 0
+        self._stats_versions = 0
+        self._stats_bytes = 0
+
+        # Initialize client
+        self.s3_client = self._create_s3_client(region_name, profile, endpoint_url)
+
+        # Validate bucket immediately
+        self._validate_bucket_versioning()
+
+    def _create_s3_client(
+        self,
+        region: str | None,
+        profile: str | None,
+        endpoint: str | None,
+    ) -> BaseClient:
+        """Create the Boto3 S3 client."""
+        session_kwargs = {}
+        client_kwargs = {}
+
+        if profile:
+            session_kwargs.update({"profile_name": profile})
+
+        if region:
+            session_kwargs.update({"region_name": region})
+
+        if endpoint:
+            client_kwargs.update({"endpoint_url": endpoint})
+
+        try:
+            session = boto3.Session(**session_kwargs)
+
+            return session.client("s3", **client_kwargs)
+        except (BotoCoreError, ClientError) as e:
+            raise RuntimeError(f"Failed to initialize S3 client: {e}") from e
+
+    def _validate_bucket_versioning(self) -> None:
+        """Ensure the bucket has versioning enabled or suspended."""
+        try:
+            response = self.s3_client.get_bucket_versioning(Bucket=self.bucket)
+            status = response.get("Status")
+
+            if status not in ("Enabled"):
+                raise RuntimeError(
+                    f"Bucket '{self.bucket}' does not have versioning history (Status: {status}). "
+                    "Nothing to scan."
+                )
+        except ClientError as e:
+            raise RuntimeError(
+                f"Could not check versioning status for '{self.bucket}': {e}"
+            ) from e
+
+        logger.debug(
+            "\nBucket: '%s'\nVersioning: %s\nRegion: %s\nEndpoint: %s\nPrefix: %s \nDeleted After: %s",
+            self.bucket,
+            status,
+            self.s3_client.meta.region_name,
+            self.s3_client.meta.endpoint_url,
+            self.prefix or "No prefix specified",
+            self.deleted_after.isoformat()
+            if self.deleted_after
+            else "No deleted after date specified",
+        )
+
+    def _print_progress(self, delta: int, label: str = "Deletable") -> None:
+        self._scanned_count += delta
+        if sys.stdout.isatty():
```

Comment:
Do not do any branching on printing based on `tty`-ness. Just print a regular log from the caller instead. Kill this function all together.

Reply:
Just print to STDOUT and you don't have to solve any of these problems. 

https://github.com/opengisch/QFieldCloud/pull/1445#discussion_r2650033258